### PR TITLE
[WNMGDS-877] Update type sizing classes and variables

### PR DESCRIPTION
**Pull request by line47**
## Summary
Added new agnostic variables and classes to replace the semantic variables and classes in the future. 
This PR adds new classes and removes the old from documentation. All old classes will still work and are still part of the codebase to prevent breaking changes. 

**Example of semantic vs agnostic**
`$h1-font-size` is a semantic variable because it refers to a specific element, the `<h1>` heading. 
`$font-size-xl` is an agnostic variable because it does not refer to a specific element.
`.ds-u-font-size--h1` is a semantic class because it refers to a specific element, the `<h1>` heading. 
`.ds-u-font-size--xl` is an agnostic class because it does not refer to a specific element.

Switching to agnostic naming is less confusing because you don't end up with code that uses a class that refers to a different element. For example, in the code below we have an `<h1>` element that uses a class name that refers to `<h3>`
```
<h1 class="ds-h3">Heading one that is styled with H3</h1>
```

### Added
- Added agnostic font size variables 
    Using current variables for values to ensure nothing breaks 
```
$font-size-5xl: $display-font-size !default; //60px
$font-size-4xl: $title-font-size !default; // 48px
$font-size-3xl: $h1-font-size !default; // 36px - h1
$font-size-2xl: $h2-font-size !default; // 24px - h2
$font-size-xl: $h3-font-size !default; // 21px - h3
$font-size-lg: $lead-font-size !default; // 18px - h4 - lead - body large
$font-size-md: $base-font-size !default; // 16px - h5 - base - body medium
$font-size-sm: $small-font-size !default; // 14px - h6 - body small
```
- Added agnostic font-size utility class names
   
```
.ds-u-font-size--[display|title|h1|h2|h3|h4|lead|base|small]
```

### Changed
- Updated documentation pages for typography and font-sizing 
- Updated type examples to use new agnostic class names
- Updated variables in components to use agnostic variable names for font sizing

### Deprecated
With this PR H tag variable and class names are now marked for deprecation in the future in favor of the new agnostic naming convention. 

## How to test
Check out the Demo URL pages 
- [Typography](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/styles/typography/)
- [Font size utility](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/utilities/font-size/)

- Test this branch on both the Mgov and HCgov child design systems by updating the package.json to point to this branch. There should be no visible issue with font sizing.

### CODE REVIEW

**Review by LouisFettet — COMMENTED** [review]
I pulled these changes into the SEP Screener, and for the most part it looks great! I noticed a few issues with backwards-compatibility which I've noted below.

File: packages/design-system-docs/src/styles/components/_Frame.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-$_link-font-size: $small-font-size;
+$_link-font-size: $font-size-md;
```
[the remark this answers is not in the document]
Comment (line47) [reply packages/design-system-docs/src/styles/components/_Frame.scss:1]: Nope, thanks for the catch!

**Comment by line47** [comment]
Thanks for the review @LouisFettet 🥳  🙇

**Review by LouisFettet — APPROVED** [review]
LGTM, I'm not seeing any backwards-compatibility issues with SEP Screener.

**Review by nichia — COMMENTED** [review]
New variables & classes looks good. 
Just a thought - for continual support of the existing variables/classes (until they get deprecated) maybe continue to include them on the document site (for example `.ds-u-font-size--5xl` replacing `.ds-display`) until these get fully deprecated

**Comment by line47** [comment]
That's a good point @nichia I was thinking going forward we don't want people to use the old classes so that's why I changed them in the documentation.

**Review by kimieann — APPROVED** [review]
LGTM. I do like Ni's idea of adding a guide of how to convert from the old to the new, but everything is working

**Comment by line47** [comment]
Updated to add some guidance for updating to the new classes 
- http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/startup/theming/
- http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/styles/typography/
- http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/utilities/font-size/

File: packages/design-system-docs/src/pages/startup/theming.md [hunk 2]
```diff
@@ -105,3 +105,65 @@ If you're not using Sass, another way to customize the design system is by overr
...
+We've added new variables to be used to customize the global sizing values.
+
+- `$font-size-5xl`
+- `$font-size-4xl`
+- `$font-size-3xl`
+- `$font-size-2xl`
+- `$font-size-xl`
+- `$font-size-lg`
+- `$font-size-md`
+- `\$font-size-sm
```
Comment (nichia) [inline packages/design-system-docs/src/pages/startup/theming.md:169]:
```suggestion
- `$font-size-sm`
```

**Review by nichia — APPROVED** [review]
🚀

**Comment by kimieann** [comment]
@nichia @line47 Since Scott is out today I went ahead and added a commit with Ni's suggested change to fix the appearance on theming.md https://github.com/CMSgov/design-system/pull/992/commits/de8ad20f4e61471ce8e66c840b63f2c0ed32181f

Will go ahead and merge once tests pass.

**Pull request merged**